### ISSUE TITLE
Cannot install it

### ISSUE THREAD

**Issue by Dmunch04** [marcobambini/gravity#271]
Whenever I try to run the `make` command in the directory, it gives me this error:

`cc -Isrc/compiler/ -Isrc/runtime/ -Isrc/shared/ -Isrc/utils/ -Isrc/optionals/ -std=gnu99 -fgnu89-inline -fPIC -DBUILD_GRAVITY_API -O2   -c -o src/compiler/gravity_ast.o src/compiler/gravity_ast.c
process_begin: CreateProcess(NULL, cc -Isrc/compiler/ -Isrc/runtime/ -Isrc/shared/ -Isrc/utils/ -Isrc/optionals/ -std=gnu99 -fgnu89-inline -fPIC -DBUILD_GRAVITY_API -O2 -c -o src/compiler/gravity_ast.o src/compiler/gravity_ast.c, ...) failed.
make (e=2): Den angivne fil blev ikke fundet.
make: *** [src/compiler/gravity_ast.o] Fejl 2`

Please help me

**Comment by marcobambini** [comment]
Which compiler are you using? I just re-tried on my machine and everything worked as expected.

**Comment by Dmunch04** [comment]
I honestly have no idea. For some time ago, I wanted to install a language. They told me to either use g++ or make. So I installed make

**Comment by marcobambini** [comment]
gcc -v from the terminal should give you the exact version.
Which operating system are you using?

**Comment by Dmunch04** [comment]
I'm using Windows 10 PRO 64-bit

Output of `gcc-v`:
`Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=D:/Program\ Files/TDM-GCC-64/bin/../libexec/gcc/x86_64-w64-mingw32/5.1.0/lto-wrapper.exe
Target: x86_64-w64-mingw32
Configured with: ../../../src/gcc-5.1.0/configure --build=x86_64-w64-mingw32 --enable-targets=all --enable-languages=ada,c,c++,fortran,lto,objc,obj-c++ --enable-libgomp --enable-lto --enable-graphite --enable-cxx-flags=-DWINPTHREAD_STATIC --disable-build-with-cxx --disable-build-poststage1-with-cxx --enable-libstdcxx-debug --enable-threads=posix --enable-version-specific-runtime-libs --enable-fully-dynamic-string --enable-libstdcxx-threads --enable-libstdcxx-time --with-gnu-ld --disable-werror --disable-nls --disable-win32-registry --prefix=/mingw64tdm --with-local-prefix=/mingw64tdm --with-pkgversion=tdm64-1 --with-bugurl=http://tdm-gcc.tdragon.net/bugs
Thread model: posix
gcc version 5.1.0 (tdm64-1)`

**Comment by IngwiePhoenix** [comment]
I will try to reproduce this behaviour, but looking at this bit:

```
(...snip...)
process_begin: CreateProcess(NULL, cc -Isrc/compiler/ -Isrc/runtime/ -Isrc/shared/ -Isrc/utils/ -Isrc/optionals/ -std=gnu99 -fgnu89-inline -fPIC -DBUILD_GRAVITY_API -O2 -c -o src/compiler/gravity_ast.o src/compiler/gravity_ast.c, ...) failed. make (e=2): Den angivne fil blev ikke fundet. make: *** [src/compiler/gravity_ast.o] Fejl 2
```

I see the `(e=2)` error message. This usually is a I/O related message. Now, I dont speak that language used in the error message, so it does look cryptic to me ^^; But, so far, `e=2` can either mean:

- The file being accessed or that is supposed to be written to, is damaged.
- There may not be enough space on disk.
- There may not be enough RAM.

Could you tell me what language your OS is set to so I can use Google translate? Or give me a translation of the message instead?

I need this bit: `Den angivne fil blev ikke fundet.`

**Comment by Dmunch04** [comment]
The file may be damaged, although I copied it straight from the repo. And I for sure have enough disk space and ram. The OS language is Danish. And that text line means: `The specified file were not found`, or at least something close to that

**Comment by Dmunch04** [comment]
Or probably given instead of specified

**Comment by IngwiePhoenix** [comment]
So for some reason, the file `src/compiler/gravity_ast.c` can not be found. Weird? Try a fresh clone and try to re-run make. Alternatively, see if you can get CMake to work.

Also what GCC did you install? From the output of `gcc -v`, I see the name `TDM-GCC-64`. It apparently is a MinGW 64-bit variant. Never heared of that GCC install before - well, I did hear about MinGW, but not about TDM-GCC. Can you give me any info on that one? I'll be installing this to see if I can reproduce what happened.

Also, which OS are you using? I can set up either Win7 or Win10 - 32 and 64 bit. So for a proper reproduction, I need to know what OS you are using. :)

**Comment by Dmunch04** [comment]
I'll try and do a fresh clone when I get home then. I can send you the link to which I installed. I am on Win10-64-PRO

**Comment by Dmunch04** [comment]
Nevermind. I can't find the link. But which are you using?

**Comment by IngwiePhoenix** [comment]
On Windows, this: https://mingw-w64.org/doku.php
But I mainly work on macOS and Linux...where GCC (or LLVM Clang) are pretty much native. Its a bit tricky under Windows. :)

**Comment by IngwiePhoenix** [comment]
I just did a cross-compile with MinGW for a 64-bit Windows.

@marcobambini If you are curios on how I did that, I will also attach my toolchain file that I used. A resource from where I got the toolchain file itself is here: https://metricpanda.com/rival-fortress-update-11-cross-compiling-for-three-platforms

I just made my own system fit, hence I did not need to adjust the file itself.


[mingw-tc.cmake.txt](https://github.com/marcobambini/gravity/files/3079692/mingw-tc.cmake.txt)
(remove `.txt`)

[gravity.exe.gz](https://github.com/marcobambini/gravity/files/3079694/gravity.exe.gz)
(Either use the `gunzip` tool or any other unarchiver that supports GZip)
